File: auth-service/src/main/java/com/masarnovsky/popugjira/auth/controller/AccountController.java
package com.masarnovsky.popugjira.auth.controller;

import com.masarnovsky.popugjira.auth.model.Account;
import com.masarnovsky.popugjira.auth.service.AccountService;
import lombok.AllArgsConstructor;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/authentication/accounts")
@AllArgsConstructor
public class AccountController {

    private final AccountService accountService;

    @GetMapping
    public List<Account> findAll() {
        return accountService.findAll();
    }

    @PostMapping
    public Account save(@RequestBody Account account) {
        return accountService.save(account);
    }

    @GetMapping("/{publicId}")
    public Account findByPublicId(@PathVariable String publicId) {
        return accountService.findByPublicId(publicId);
    }
}

File: tasks-service/src/main/kotlin/com/masarnovsky/popugjira/tasks/controller/AccountController.kt
package com.masarnovsky.popugjira.tasks.controller

import com.masarnovsky.popugjira.tasks.model.Account
import com.masarnovsky.popugjira.tasks.model.AccountDto
import com.masarnovsky.popugjira.tasks.service.AccountService
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/accounts")
class AccountController(
        private val accountService: AccountService
) {

    @GetMapping("/{id}")
    fun findById(@PathVariable("id") id: String): Account {
        return accountService.findByPublicId(id)
    }

    @PostMapping
    fun createAccount(@RequestBody request: AccountDto): Account {
        return accountService.save(request)
    }

    @GetMapping
    fun findAll(): List<Account> {
        return accountService.findAll()
    }

    @DeleteMapping
    fun clear() {
        accountService.clear()
    }
}